docs/BUILD.md

```diff
@@ -0,0 +1,163 @@
+* [html-reporter](https://www.npmjs.com/package/html-reporter): plugin for Gemini to produce an HTML report of the CSS regression tests.
+* [surge](https://surge.sh/): for publishing static website. We use this to publish the kitchen sink app.
+
+## Additional NPM Scripts
+
+There are a few other NPM scripts that can be useful during build and development.
+
+##### `npm start`
+
+The start command starts up our demo app using the Angular CLI on port 4200 and watches for file changes for live reload.
+
+##### `npm run build`
+
+This script builds npm package candidates for all three packages we currently publish: `@clr/angular`, `@clr/ui`, and
+`@clr/icons` under the `/dist` folder.
+
+##### `npm test` and `npm run angular:test:watch`
+
+The `test` script runs the unit tests using karma. The entry point for the tests is `tests/tests.entry.ts`.
+You may locally modify this file to constrain which tests to run if you are testing for specific components and don't want
+to run all the tests. You can run the tests in watch mode, so they run continuously `npm run angular:test:watch`.
+
+##### `npm run build:ci`
+
+The `build:ci` script is used by Travis-CI to run all of the checks, such as format, lint, and unit tests.
+If the code doesn't pass both the format and lint checkers, then the build fails before running the unit tests.
+
+##### `npm run format` and `npm run format:fix`
+
+We use [clang-format](https://github.com/angular/clang-format) as formatter for our code. `npm run format` will not
+actually format the file but only check if there are any files that would be changed. We do this via a shell script (`./scripts/clang-check.sh`),
+which runs the clang-format command (with `-output-replacements-xml` flag) and greps for any replacement that would be produced.
+
+The `npm run format:fix` does the actual formatting according to the rules specified in `.clang-format` file.
+
+##### `npm run angular:build`
+
+This script produces the `@clr/angular` package using [ng-packagr](https://github.com/dherges/ng-packagr).
+
+The script copies over the `package.json` template from our `npm` folder (this contains templates for `package.json` and
+`README.md` for all of our packages) into `src/clr-angular` and sets the correct version number. This step is necessary
+because `ng-packagr` requires the `package.json` to be at the root of the `src` (defined in `ng-package.json`).
+
+##### `npm run icons:build`
+
+This script produces the `clr-icons` package by bundling js files that is included in the consuming app.
+The `post` script generates the SVG files and also zipped up files for the icon sets. Note that this script partially
+relies on `webpack` as well, since the `webpack` script produces the `clr-icons.css` and `clr-icons.min.css` files.
+The `webpack` script also processes the `package.json` and `README.md` files for all of our packages.
+This means that running `npm icons:build` by itself does NOT produce a complete package.
+
```

Comment:
Should we add a section about `npm run core: build`? I know it's still a work in progress, but I assume we should add it since other parts of our doc mentions `core` as a deliverable.

Reply:
I'm ok with creating a separate issue for this if you feel that we should finalize details rather than adding in this PR.